core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -75,6 +75,20 @@ private TableProperties() {
    */
   public static final String CURRENT_SNAPSHOT_TIMESTAMP = "current-snapshot-timestamp-ms";
 
+  /**
+   * Reserved table property for default partition spec.
+   * <p>
+   * This reserved property is used to store the default partition spec.
+   */
+  public static final String DEFAULT_PARTITION_SPEC = "default-partition-spec";
```

Comment:
It might be helpful to clarify what is meant by `default` for both of these.

Each javadoc states that it’s for the “default” spec / sort order, but I’m still not entirely sure what that means.

Is it the current sort order / partition spec that would be used if the user doesn’t override it for an individual query ?

Reply:
Maybe something like “JSON representation of the table’s current configured partition spec, which will be used if not overridden for individual writes”. Kind of wordy but something along those lines would be helpful for me if quickly looking through the JavaDocs etc. Will leave that decision to you though.

Reply:
Yes, it is for the current partition spec and sort order. Make sense to me. Will make the change.

Reply:
I think you reverted too many?  Should be 'current'.

Reply:
I was trying to be consistent with the name in metadata.json, which is `default-partition-spec`. The same for sort order.  I changed the comments though.

Reply:
OK I see, yea I always find it confusing. 

In the comment, maybe we can add that they are equivalent, otherwise the comment is even more confusing:
```
Reserved table property for the JSON representation of current (default) schema.
```

Reply:
It is confusing. I like the `current` more. I keep the original name just for consistency.